Services/GayboardService.cs

```diff
@@ -14,45 +14,85 @@ public class GayboardService : DiscordBotService {
 
     protected override async ValueTask OnReactionAdded(ReactionAddedEventArgs e) {
         var message = await Client.FetchMessageAsync(e.ChannelId, e.MessageId) as IUserMessage;
-        if (message == null)
-            return;
-
-        if (message.Reactions.TryGetValue(out IReadOnlyDictionary<IEmoji, IMessageReaction>? reactions)) {
-            if (reactions.TryGetValue(DETECTABLE_EMOTE, out IMessageReaction? reaction) && reaction.Count == MIN_REACTIONS_REQUIRED) {
-                IReadOnlyList<IMessage> oldMessages = await Client.FetchMessagesAsync(CHANNEL_ID, limit: 50);
-                string contentString = $"https://discord.com/channels/{e.GuildId}/{message.ChannelId}/{message.Id}";
-                if (oldMessages.Any(x => x.Content.Equals(contentString))) {
-                    Logger.LogWarning("Failed to post this gay message as it was already posted");
-                    return;
-                }
-
-                var embed = new LocalEmbed {
-                    Color = Color.Pink,
-                    Author = new LocalEmbedAuthor {
-                        Name = message.Author.Name,
-                        IconUrl = message.Author.GetAvatarUrl()
-                    },
-                    Description = message.Content,
-                    Footer = new LocalEmbedFooter {
-                        Text = TimeZoneInfo.ConvertTime(message.CreatedAt(), SchedulerService.GetTimeZoneInfo()).ToString("dd/MM/yyyy HH:mm"),
-                        IconUrl ="https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/LGBT_Rainbow_Flag.png/800px-LGBT_Rainbow_Flag.png"
-                    }
-                };
-
-                if (message.Attachments.Count > 0) {
-                    IAttachment firstAttachment = message.Attachments[0];
-                    if (firstAttachment.ContentType?.StartsWith("image/") == true) {
-                        embed.ImageUrl = firstAttachment.Url;
-                    }
-                }
-
-                var newMessage = new LocalMessage {
-                    Embeds = new List<LocalEmbed> { embed },
-                    Content = contentString
-                };
-
-                await Bot.SendMessageAsync(CHANNEL_ID, newMessage);
+        if (message == null) return;
+
+        if (!HasRequiredReactions(message)) return;
+        if (await IsAlreadyPosted(e.GuildId.GetValueOrDefault(), message)) return;
+
+        List<LocalEmbed> embeds = CreateEmbeds(message);
+        await PostToBoard(e.GuildId.GetValueOrDefault(), message, embeds);
+    }
+
+    bool HasRequiredReactions(IUserMessage message) {
+        if (!message.Reactions.TryGetValue(out IReadOnlyDictionary<IEmoji, IMessageReaction>? reactions))
+            return false;
+
+        return reactions.TryGetValue(DETECTABLE_EMOTE, out IMessageReaction? reaction)
+            && reaction.Count >= MIN_REACTIONS_REQUIRED;
```

Comment:
(Not-So-Major )Issue with not using strict equality means that the whole IsAlreadyPosted() will be called (along with the fetching) for nothing

Reply:
Fixed in 72fb59d